src/main/java/org/fundacionjala/coding/kevincristhian/CesarCrypto.java

```diff
@@ -0,0 +1,139 @@
+        int posKey = 0;
+        int sizeKey = key.length() - 1;
+        StringBuffer newKey = new StringBuffer();
+        for (int i = 0; i < message.length(); i++) {
+            if (message.charAt(i) != ' ') {
+                newKey.append(key.charAt(posKey));
+                if (posKey >= sizeKey) {
+                    posKey = 0;
+                } else {
+                    posKey++;
+                }
+            } else {
+                newKey.append(message.charAt(i));
+            }
+        }
+        return newKey.toString();
+    }
+
+    /**
+     * encodeVigenere vigenere.
+     * @param message message to encodeVigenere.
+     * @param keyWord word.
+     * @return string.
+     */
+    public String encodeVigenere(final String message, final String keyWord) {
+        StringBuffer newMessage = new StringBuffer();
+        if (message != null) {
+            int size = message.length();
+            String newKeyWord;
+            newKeyWord = createKeyString(message, keyWord);
+            char newChar;
+            for (int i = 0; i < size; i++) {
+                if (message.charAt(i) >= 'A' && message.charAt(i) <= 'Z') {
+                    newChar = (char) (message.charAt(i) + newKeyWord.charAt(i) - SIXTY_FOUR);
+                    if (newChar > 'Z') {
+                        newChar = (char) ((newChar - NINETY) + SIXTY_FOUR);
+                    }
+                } else {
+                    newChar = message.charAt(i);
+                }
+                newMessage.append(newChar);
+            }
+        }
+        return newMessage.toString();
+    }
+
+    /**
+     * decode vigenere.
+     * @param message to decode.
+     * @param keyWord key word.
+     * @return string
+     */
+    public String decodeVigenere(final String message, final String keyWord) {
+        StringBuffer newMessage = new StringBuffer();
+        if (message != null) {
+            int size = message.length();
+            String newKeyWord;
+            newKeyWord = createKeyString(message, keyWord);
+            char newChar;
+            for (int i = 0; i < size; i++) {
+                if (message.charAt(i) >= 'A' && message.charAt(i) <= 'Z') {
+                    newChar = (char) (message.charAt(i) - newKeyWord.charAt(i) + SIXTY_FOUR);
+                    if (newChar < 'A') {
+                        newChar = (char) ((newChar + NINETY) - SIXTY_FOUR);
+                    }
+                } else {
+                    newChar = message.charAt(i);
+                }
+                newMessage.append(newChar);
+            }
+        }
+        return newMessage.toString();
+    }
+    /**
+     * decode cesar (it can be used to encodeVigenere too).
+     * @param message message to decode.
+     * @param key key to decode.
+     * @return string.
+     */
+    public String decode(final String message, int key) {
+        StringBuffer newMessage = new StringBuffer();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)